src/platform/macos/view.rs

```diff
@@ -436,6 +475,7 @@ extern fn key_down(this: &Object, _sel: Sel, event: id) {
             },
         };
 
+        // NOTE: this code is run twice.
```

Comment:
Why is this code run twice?

Reply:
IMO it doesn't have a good reason to be run twice.  The ~10 lines below the comment here seem to be lifted entirely from the `get_characters()` method, which is used to set `state.raw_characters` earlier.  My best guess is that this has happened as a somewhat lazy way to bring `string` into scope without dealing with `Option`/ownership.
Also, does `get_characters()` need to return an `Option` when it only ever seems to return `Some()`?

I'm not sure on the etiquette here.  I wanted to fix this but wasn't entirely sure what the best way to go about it would be.  Thoughts?

Reply:
Oh, I see. I was confused and thought the comment was about a change you made.

Yeah, this redundancy is definitely silly. Also, IIRC `get_characters` came later: it was something I nagged someone into making to simplify their PR, but I didn't nag them into using it everywhere... there's a lot of room for refactoring here, and it would be very much appreciated!

Um, I can't remember any reason for `get_characters` returning an `Option`, so that's pretty funny. While we could keep it this way and check for `nil`, it's probably fine to just use `assert_ne!(characters, nil)` and then always return `String`.